src/terminal/adapter/SixelParser.cpp

```diff
@@ -610,7 +610,9 @@ void SixelParser::_updateTextColors()
     // the text output as well.
     if (_conformanceLevel <= 3 && _maxColors > 2 && _colorTableChanged) [[unlikely]]
     {
-        for (IndexType tableIndex = 0; tableIndex < _maxColors; tableIndex++)
+        // _maxColors is 64-bit and tableIndex is 8-bit. If _maxColors is higher
+        // than the 8-bit integer limit, we will loop indefinitely.
+        for (IndexType tableIndex = 0; tableIndex < (saturated_cast<uint8_t>(_maxColors)); tableIndex++)

```

Comment:
In this situation `_maxColors` will never be greater than 16, because that's the maximum value for conformance level 3 (which we're checking in the condition above). Would it satisfy the CodeQL scan if we also added a `_maxColors <= 16` check to that condition? That seems like a preferable fix to me, because it makes it clear what values we're actually dealing with here. Suggesting that `_maxColors` has a 64-bit range here is very misleading.